### DISCUSSION TITLE
AI zukijourney error(Array must end with a user role)

### DISCUSSION THREAD

**Discussion by gitttyyy** [end-4/dots-hyprland#305] [Q&A] [answered]
```json
{
    "error": {
        "message": "Array must end with a user role.",
        "type": "error",
        "param": null,
        "code": 469
    }
}
```

I`m not good at js,how can I fix it?

**Comment by end-4** [comment, marked as answer]
https://github.com/end-4/dots-hyprland/commit/689b16b90acad4aefb4f30f4223750d8cf3e4e4f 
does this work? zukijourney locks api key to ip and the detection for it sucks so i can't test it

**Reply by gitttyyy** [reply]
Ok,is work,but have some problem in style
![图片](https://github.com/end-4/dots-hyprland/assets/154318356/e71c1a39-0a3f-44f9-9857-86242a815aca)
Also have a problem that  I can't send a message through enter or ctrl+enter, I can only click the send button, which is a bit unreasonable.

**Reply by end-4** [reply]
does this happen when you don't have it respond in chinese? weird characters seem to mess up markup parsing

**Reply by end-4** [reply]
> Also have a problem that I can't send a message through enter or ctrl+enter, I can only click the send button, which is a bit unreasonable.

Update hyprland

**Comment by gitttyyy** [comment]
> > Also have a problem that I can't send a message through enter or ctrl+enter, I can only click the send button, which is a bit unreasonable.
> 
> Update hyprland

What this mean? I just recently started using this amazing dots and changed some keybinding and env based on my own,I not find what make this happen

**Reply by end-4** [reply]
i'that one time when it didn't work for me was when i use an old hyprland version. not really sure.

**Reply by gitttyyy** [reply]
I'm always up to date🐶

**Comment by gitttyyy** [comment]
> does this happen when you don't have it respond in chinese? weird characters seem to mess up markup parsing

It`s only happen when use chinese
